storm-server/src/main/java/org/apache/storm/localizer/AsyncLocalizer.java

```diff
@@ -456,7 +459,28 @@ public void releaseSlotFor(LocalAssignment assignment, int port) throws IOExcept
             topoConfBlob.removeReference(pna);
         }
 
-        for (LocalResource lr : getLocalResources(pna)) {
+
+        // ALERT: A possible race condition could be resolved by separating the thread pools into downloadExecService and taskExecService
+        // https://git.ouroath.com/storm/storm/commit/ebd52b37c7448d381d31451e46e8f19c6e51352d#diff-74535cb89e9e926ad424a8d1e2fa9586
+        // Will need further investigation if the race condition happens again
+        List<LocalResource> localResources;
+        try {
+            // Precondition1: Base blob stormconf.ser and stormcode.ser have been localized
+            // Precondition2: Both these two blob files are fully downloaded and proper permission been set
+            localResources = getLocalResources(pna);
+        } catch (FileNotFoundException fnfException) {
+            localResourceFileNotFound.mark();
+            LOG.warn("Local base blobs have not been downloaded yet. "
+                        + "DownloadExecService is too busy", fnfException);
+            LOG.info("Port and assignment info: {}", pna);
+            return;
+        } catch (IOException ioException) {
+            LOG.error("Unable to read local conf file. ", ioException);
+            LOG.info("Port and assignment info: {}", pna);
```

Comment:
`LOG.info("Port and assignment info: {}", pna); ` are repeated twice here. We can write it in the following way:

```
catch (IOException e) {
            //common1
            if (e instanceof FileNotFoundException) {
                //xx
            } else {
                //yy
            }
       // common2
        }
```

Reply:
changed